src/prognostic_equations/forcing/external_forcing.jl

```diff
@@ -5,12 +5,75 @@
 import Thermodynamics as TD
 import ClimaCore.Spaces as Spaces
 import ClimaCore.Fields as Fields
+import NCDatasets as NC
+import StatsBase
+import Dierckx
+
+
+"""
+    get_nc_data(ds::NCDatasets.Dataset, var::String)
+
+Returns the data for variable `var`.
+"""
+function get_nc_data(ds, var::String)
+    key_options = (var,)
+    for key in key_options
+        if haskey(ds, key)
+            return ds[key]
+        else
+            for group_option in ["profiles", "reference", "timeseries"]
+                haskey(ds.group, group_option) || continue
+                if haskey(ds.group[group_option], key)
+                    return ds.group[group_option][key]
+                end
+            end
+        end
+    end
+    return nothing
+end
+
+
+function pyinterp(x, xp, fp)
+    spl = Dierckx.Spline1D(xp, fp; k = 1)
+    return spl(vec(x))
+end
```

Comment:
could we either rename this function (don't understand why it's called `pyinterp`, or just write out the spline in `getvar` and `get_nudgevar` below?